------------------------------------
1. TDV 8.3 Cannot connect on Linux to localhost:9830 (TDV 8.3)
------------------------------------
./ExecutePDTool.sh -exec ../resources/plans/UnitTest-Resource-exists.dp    -config deploy_GIT_9830.properties

2020-10-22 16:05:19,808 main INFO [com.tibco.ps.deploytool.DeployManagerUtil] - <-------------------------------------------------------->
2020-10-22 16:05:19,808 main INFO [com.tibco.ps.deploytool.DeployManagerUtil] - <------------------ DEPLOYMENT MANAGER ------------------>
2020-10-22 16:05:19,808 main INFO [com.tibco.ps.deploytool.DeployManagerUtil] - <-------------------------------------------------------->
2020-10-22 16:05:19,808 main INFO [com.tibco.ps.deploytool.DeployManagerUtil] - <Environment variable CIS_VERSION=8.3.0>
2020-10-22 16:05:19,809 main INFO [com.tibco.ps.deploytool.DeployManagerUtil] - <Set environment variable PDTOOL_ADAPTER_CONFIG_PATH=/opt/TIBCO/PDTool8.3.0_GIT/PDTool/resources/config/8.3.0/cis_adapter_config.xml>
2020-10-22 16:05:19,809 main INFO [com.tibco.ps.deploytool.DeployManagerUtil] - <Calling Action doResourceExist>
2020-10-22 16:05:21,200 main ERROR [com.tibco.ps.deploytool.DeployManagerUtil] - <Error while invoking method doResourceExist>
java.lang.reflect.InvocationTargetException
        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
        at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.base/java.lang.reflect.Method.invoke(Method.java:566)
        at com.tibco.ps.deploytool.DeployManagerUtil.main(DeployManagerUtil.java:120)
        at com.tibco.ps.deploytool.CisDeployTool.executeAction(CisDeployTool.java:675)
        at com.tibco.ps.deploytool.CisDeployTool.execCisDeployTool(CisDeployTool.java:358)
        at com.tibco.ps.deploytool.DeployManagerImpl.execCisDeployTool(DeployManagerImpl.java:75)
        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
        at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.base/java.lang.reflect.Method.invoke(Method.java:566)
        at com.tibco.ps.deploytool.DeployManagerUtil.main(DeployManagerUtil.java:120)
Caused by: javax.xml.ws.WebServiceException: java.lang.RuntimeException: java.lang.IllegalArgumentException: Illegal base64 character 20
        at com.sun.xml.ws.transport.http.client.HttpTransportPipe.process(HttpTransportPipe.java:269)
        at com.sun.xml.ws.transport.http.client.HttpTransportPipe.processRequest(HttpTransportPipe.java:184)
        at com.sun.xml.ws.transport.DeferredTransportPipe.processRequest(DeferredTransportPipe.java:109)
        at com.sun.xml.ws.api.pipe.Fiber.__doRun(Fiber.java:641)
        at com.sun.xml.ws.api.pipe.Fiber._doRun(Fiber.java:600)
        at com.sun.xml.ws.api.pipe.Fiber.doRun(Fiber.java:585)
        at com.sun.xml.ws.api.pipe.Fiber.runSync(Fiber.java:482)
        at com.sun.xml.ws.client.Stub.process(Stub.java:323)
        at com.sun.xml.ws.client.sei.SEIStub.doProcess(SEIStub.java:161)
        at com.sun.xml.ws.client.sei.SyncMethodHandler.invoke(SyncMethodHandler.java:113)
        at com.sun.xml.ws.client.sei.SyncMethodHandler.invoke(SyncMethodHandler.java:93)
        at com.sun.xml.ws.client.sei.SEIStub.invoke(SEIStub.java:144)
        at com.sun.proxy.$Proxy44.getAllResourcesByPath(Unknown Source)
        at com.tibco.ps.deploytool.dao.wsapi.ResourceWSDAOImpl.getResource(ResourceWSDAOImpl.java:602)
        at com.tibco.ps.deploytool.services.ResourceManagerImpl.doResourceExist(ResourceManagerImpl.java:681)
        at com.tibco.ps.deploytool.DeployManagerImpl.doResourceExist(DeployManagerImpl.java:568)
        ... 13 more
Caused by: java.lang.RuntimeException: java.lang.IllegalArgumentException: Illegal base64 character 20
        at java.base/sun.net.www.protocol.http.HttpURLConnection.getInputStream0(HttpURLConnection.java:1529)
        at java.base/sun.net.www.protocol.http.HttpURLConnection.getInputStream(HttpURLConnection.java:1515)
        at java.base/sun.net.www.protocol.http.HttpURLConnection.getHeaderField(HttpURLConnection.java:3130)
        at java.base/java.net.HttpURLConnection.getResponseCode(HttpURLConnection.java:536)
        at com.sun.xml.ws.transport.http.client.HttpClientTransport.readResponseCodeAndMessage(HttpClientTransport.java:207)
        at com.sun.xml.ws.transport.http.client.HttpTransportPipe.createResponsePacket(HttpTransportPipe.java:274)
        at com.sun.xml.ws.transport.http.client.HttpTransportPipe.process(HttpTransportPipe.java:265)
        ... 28 more
Caused by: java.lang.IllegalArgumentException: Illegal base64 character 20
        at java.base/java.util.Base64$Decoder.decode0(Base64.java:743)
        at java.base/java.util.Base64$Decoder.decode(Base64.java:535)
        at java.base/java.util.Base64$Decoder.decode(Base64.java:558)
        at java.base/sun.net.www.protocol.http.ntlm.NTLMAuthentication.buildType3Msg(NTLMAuthentication.java:258)
        at java.base/sun.net.www.protocol.http.ntlm.NTLMAuthentication.setHeaders(NTLMAuthentication.java:236)
        at java.base/sun.net.www.protocol.http.HttpURLConnection.getInputStream0(HttpURLConnection.java:1773)
        at java.base/sun.net.www.protocol.http.HttpURLConnection.getInputStream(HttpURLConnection.java:1515)
        at java.base/java.net.HttpURLConnection.getResponseCode(HttpURLConnection.java:527)
        ... 31 more


------------------------------------
2. TDV 8.2 Cannot connect to from Liniux to mtinius-x1e2:9800 (TDV 8.2)
------------------------------------
./ExecutePDTool.sh -exec ../resources/plans/UnitTest-Resource-exists.dp    -config deploy_GIT_9800.properties

2020-10-22 16:01:22,247 main INFO [com.tibco.ps.deploytool.DeployManagerUtil] - <-------------------------------------------------------->
2020-10-22 16:01:22,247 main INFO [com.tibco.ps.deploytool.DeployManagerUtil] - <------------------ DEPLOYMENT MANAGER ------------------>
2020-10-22 16:01:22,247 main INFO [com.tibco.ps.deploytool.DeployManagerUtil] - <-------------------------------------------------------->
2020-10-22 16:01:22,247 main INFO [com.tibco.ps.deploytool.DeployManagerUtil] - <Environment variable CIS_VERSION=8.0.0>
2020-10-22 16:01:22,248 main INFO [com.tibco.ps.deploytool.DeployManagerUtil] - <Set environment variable PDTOOL_ADAPTER_CONFIG_PATH=/opt/TIBCO/PDTool8.0.0_GIT/PDTool/resources/config/8.0.0/cis_adapter_config.xml>
2020-10-22 16:01:22,248 main INFO [com.tibco.ps.deploytool.DeployManagerUtil] - <Calling Action doResourceExist>
2020-10-22 16:01:44,882 main ERROR [com.tibco.ps.deploytool.DeployManagerUtil] - <Error while invoking method doResourceExist>
java.lang.reflect.InvocationTargetException
        at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
        at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.lang.reflect.Method.invoke(Method.java:498)
        at com.tibco.ps.deploytool.DeployManagerUtil.main(DeployManagerUtil.java:120)
        at com.tibco.ps.deploytool.CisDeployTool.executeAction(CisDeployTool.java:675)
        at com.tibco.ps.deploytool.CisDeployTool.execCisDeployTool(CisDeployTool.java:358)
        at com.tibco.ps.deploytool.DeployManagerImpl.execCisDeployTool(DeployManagerImpl.java:75)
        at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
        at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.lang.reflect.Method.invoke(Method.java:498)
        at com.tibco.ps.deploytool.DeployManagerUtil.main(DeployManagerUtil.java:120)
Caused by: com.sun.xml.ws.client.ClientTransportException: HTTP transport error: java.net.ConnectException: Connection refused (Connection refused)
        at com.sun.xml.ws.transport.http.client.HttpClientTransport.getOutput(HttpClientTransport.java:132)
        at com.sun.xml.ws.transport.http.client.HttpTransportPipe.process(HttpTransportPipe.java:256)
        at com.sun.xml.ws.transport.http.client.HttpTransportPipe.processRequest(HttpTransportPipe.java:184)
        at com.sun.xml.ws.transport.DeferredTransportPipe.processRequest(DeferredTransportPipe.java:109)
        at com.sun.xml.ws.api.pipe.Fiber.__doRun(Fiber.java:641)
        at com.sun.xml.ws.api.pipe.Fiber._doRun(Fiber.java:600)
        at com.sun.xml.ws.api.pipe.Fiber.doRun(Fiber.java:585)
        at com.sun.xml.ws.api.pipe.Fiber.runSync(Fiber.java:482)
        at com.sun.xml.ws.client.Stub.process(Stub.java:323)
        at com.sun.xml.ws.client.sei.SEIStub.doProcess(SEIStub.java:161)
        at com.sun.xml.ws.client.sei.SyncMethodHandler.invoke(SyncMethodHandler.java:113)
        at com.sun.xml.ws.client.sei.SyncMethodHandler.invoke(SyncMethodHandler.java:93)
        at com.sun.xml.ws.client.sei.SEIStub.invoke(SEIStub.java:144)
        at com.sun.proxy.$Proxy41.getAllResourcesByPath(Unknown Source)
        at com.tibco.ps.deploytool.dao.wsapi.ResourceWSDAOImpl.getResource(ResourceWSDAOImpl.java:602)
        at com.tibco.ps.deploytool.services.ResourceManagerImpl.doResourceExist(ResourceManagerImpl.java:681)
        at com.tibco.ps.deploytool.DeployManagerImpl.doResourceExist(DeployManagerImpl.java:568)
        ... 13 more
Caused by: java.net.ConnectException: Connection refused (Connection refused)
        at java.net.PlainSocketImpl.socketConnect(Native Method)
        at java.net.AbstractPlainSocketImpl.doConnect(AbstractPlainSocketImpl.java:350)
        at java.net.AbstractPlainSocketImpl.connectToAddress(AbstractPlainSocketImpl.java:206)
        at java.net.AbstractPlainSocketImpl.connect(AbstractPlainSocketImpl.java:188)
        at java.net.SocksSocketImpl.connect(SocksSocketImpl.java:392)
        at java.net.Socket.connect(Socket.java:607)
        at java.net.Socket.connect(Socket.java:556)
        at sun.net.NetworkClient.doConnect(NetworkClient.java:180)
        at sun.net.www.http.HttpClient.openServer(HttpClient.java:463)
        at sun.net.www.http.HttpClient.openServer(HttpClient.java:558)
        at sun.net.www.http.HttpClient.<init>(HttpClient.java:242)
        at sun.net.www.http.HttpClient.New(HttpClient.java:339)
        at sun.net.www.http.HttpClient.New(HttpClient.java:357)
        at sun.net.www.protocol.http.HttpURLConnection.getNewHttpClient(HttpURLConnection.java:1226)
        at sun.net.www.protocol.http.HttpURLConnection.plainConnect0(HttpURLConnection.java:1205)
        at sun.net.www.protocol.http.HttpURLConnection.plainConnect(HttpURLConnection.java:1056)
        at sun.net.www.protocol.http.HttpURLConnection.connect(HttpURLConnection.java:990)
        at sun.net.www.protocol.http.HttpURLConnection.getOutputStream0(HttpURLConnection.java:1340)
        at sun.net.www.protocol.http.HttpURLConnection.getOutputStream(HttpURLConnection.java:1315)
        at com.sun.xml.ws.transport.http.client.HttpClientTransport.getOutput(HttpClientTransport.java:120)
        ... 29 more



------------------------------------
3. Causes a null pointer exception
------------------------------------
# PDTool Module Debugging for Command-line and Ant (Applies to *ALL* Modules)
log4j.logger.com.tibco.ps.deploytool=DEBUG

java.lang.reflect.InvocationTargetException
        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
        at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.base/java.lang.reflect.Method.invoke(Method.java:566)
        at com.tibco.ps.deploytool.DeployManagerUtil.main(DeployManagerUtil.java:120)
        at com.tibco.ps.deploytool.CisDeployTool.executeAction(CisDeployTool.java:675)
        at com.tibco.ps.deploytool.CisDeployTool.execCisDeployTool(CisDeployTool.java:358)
        at com.tibco.ps.deploytool.DeployManagerImpl.execCisDeployTool(DeployManagerImpl.java:75)
        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
        at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.base/java.lang.reflect.Method.invoke(Method.java:566)
        at com.tibco.ps.deploytool.DeployManagerUtil.main(DeployManagerUtil.java:120)
Caused by: java.lang.NullPointerException
        at com.tibco.ps.deploytool.dao.wsapi.VCSWSDAOImpl.execCommandLineVCS(VCSWSDAOImpl.java:362)
        at com.tibco.ps.deploytool.services.VCSManagerImpl.vcs_checkin_checkout__vcs_checkout(VCSManagerImpl.java:5600)
        at com.tibco.ps.deploytool.services.VCSManagerImpl.checkout(VCSManagerImpl.java:3797)
        at com.tibco.ps.deploytool.services.VCSManagerImpl.vcsCheckoutImpl(VCSManagerImpl.java:394)
        at com.tibco.ps.deploytool.services.VCSManagerImpl.vcsCheckout(VCSManagerImpl.java:209)
        at com.tibco.ps.deploytool.DeployManagerImpl.vcsCheckout(DeployManagerImpl.java:989)
        ... 13 more